bittide/src/Clash/Signal/TH/Extra.hs

```diff
@@ -11,28 +11,60 @@ import GHC.Records (HasField (..))
 import Language.Haskell.TH
 
 {- | Derive instances of 'HasField' for a Signals of Records:
+
 Example:
+
+@
 data MyRecord = MyRecord { field1 :: Int, field2 :: Bool }
 deriveSignalHasFields ''MyRecord
+@
+
 This will generate the following instances:
+
+@
 instance HasField "field1" (Signal dom MyRecord) (Signal dom Int) where
   getField = fmap field1
 instance HasField "field2" (Signal dom MyRecord) (Signal dom Bool) where
   getField = fmap field2
+@
+
+The record type may also take type arguments. For instance, this type:
+
+@
+data MyRecord n = MyRecord { field1 :: BitVector n, field2 :: BitVector n (Unsigned 8) }
+@
+
+Will generate the following instances:
+
+@
+instance HasField "field1" (Signal dom (MyRecord n)) (Signal dom (BitVector n)) where
+  getField = fmap field1
+instance HasField "field2" (Signal dom (MyRecord n)) (Signal dom (Vec n (Unsigned 8))) where
+  getField = fmap field2
+@
 -}
 deriveSignalHasFields :: Name -> Q [Dec]
 deriveSignalHasFields recordName = do
   -- Extract information
   info <- reify recordName
   case info of
     -- If the name refers to a record type, generate instances for each field
-    TyConI (DataD _ _ _ _ [RecC _ fields] _) -> do
+    TyConI (DataD _ _ tvb _ [RecC _ fields] _) -> do
+      runIO $ mapM_ print tvb
```

Comment:
Remove this print statement

Reply:
Oops, forgot to remove when I was doing some debugging.